source/connection_manager.c

```diff
@@ -808,10 +849,56 @@ struct aws_http_connection_manager *aws_http_connection_manager_new(
     }
 
     if (options->proxy_options) {
+        /* if proxy options are set, ignore the environment variable */
         manager->proxy_config = aws_http_proxy_config_new_from_manager_options(allocator, options);
         if (manager->proxy_config == NULL) {
             goto on_error;
         }
+    } else {
+        struct aws_http_proxy_options *proxy_options = NULL;
+        struct aws_uri *proxy_uri = aws_mem_calloc(allocator, 1, sizeof(struct aws_uri));
+        /* HTTPS proxy preferred than HTTP proxy */
+        if (s_proxy_uri_init_from_env_variable(allocator, proxy_uri)) {
+            /* Envrionment is set but failed to parse it */
+            aws_mem_release(allocator, proxy_uri);
+            goto on_error;
+        }
+        if (proxy_uri) {
+            proxy_options = aws_mem_calloc(allocator, 1, sizeof(struct aws_http_proxy_options));
+            proxy_options->host = proxy_uri->host_name;
+            proxy_options->port = proxy_uri->port;
+            /* Support basic authentication. */
+            if (proxy_uri->password.len) {
+                /* Has no empty password set */
+                struct aws_http_proxy_strategy_basic_auth_options config = {
+                    .proxy_connection_type = AWS_HPCT_HTTP_FORWARD,
```

Comment:
forwarding Based on what?

More generally, what are the semantics of http_proxy vs. https_proxy?  How do you determine if this should use tunneling or forwarding (which doesn't get set here meaning it uses the legacy value which I doubt is correct behavior)?    Also if you're making a connection to an https proxy you need a tls context.

Reply:
Added a proxy env_var setting to the connection manager options that contains the connection type and tls options. Hmmm, but what if the tls options is not set? I checked the proxy test and seems like we never set the tls options for the proxy options?